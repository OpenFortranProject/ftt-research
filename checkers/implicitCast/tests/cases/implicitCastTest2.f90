! Test that we detect (and warn) about the
! conversion of real to integer.
program test

  INTEGER :: x
  REAL    :: y
  
  y = 42.4
  x = y + 4.0
  
  print *, x
  print *, y

end
