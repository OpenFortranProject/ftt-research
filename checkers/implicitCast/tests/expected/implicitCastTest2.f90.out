Warning: environment variable COMPASS_PARAMETERS is not set, trying to use compass_parameters of the current path
 Running Prerequisite SgProject
ImplicitCast: /home/dagit/research/checkers/implicitCast/tests/cases/implicitCastTest2.f90:9.2: Implicit cast in Fortran code found:  expr 'y + 4.0' from real_4 to integer_4
